packages/cli/core/src/commands/ui/deploy.ts

```diff
@@ -159,14 +160,47 @@ export default class Deploy extends CLICommand {
     this.validateDeployConfigJson(deployConfigPath, deployConfig);
     const hostedPage = this.buildHostedPage(deployConfig);
 
-    if (flags.pageId) {
-      this.updatePage({...hostedPage, id: flags.pageId});
+    const pageId = flags.pageId || (await this.getPageIdFromPage(hostedPage));
+
+    if (pageId) {
+      await this.updatePage({...hostedPage, id: pageId});
       return;
     }
 
     await this.createPage(hostedPage);
   }
 
+  private async getPageIdFromPage(
+    hostedPage: New<HostedPage, null>
+  ): Promise<string | null> {
+    const {flags} = await this.parse(Deploy);
+    const orgId = getTargetOrg(this.configuration, flags.organization);
+    const client = await this.createClient();
+    const response = await client.hostedPages.list({
+      filter: hostedPage.name,
+      perPage: 1000,
+    });
+    const page = response.items
+      .filter((page) => page.name === hostedPage.name)
+      .pop();
+
+    if (!page) {
+      return null;
+    }
+    if (await this.confirmOverwrite(page.id, orgId)) {
+      return page.id;
+    } else {
+      this.error(
+        `Page name must be unique, try changing the "name" field in "coveo.deploy.json".`
+      );
+    }
+  }
+
+  private async confirmOverwrite(pageId: string, orgId: string) {
+    return await CliUx.ux.confirm(dedent`
+      Overwrite page with ID: "${pageId}" in organization ${orgId}?`);
```

Comment:
add `? (y/n)` at the end of the message since it is not added by CliUx.

Reply:
addressed in 7259ddecceafa4d8d8e00001778ae8d29389fe56